docs/design/mono/mobile-runtimeconfig-json.md

```diff
@@ -0,0 +1,97 @@
+# Mobile runtimeconfig.json host configuration
+
+## Motivation
+
+`runtimeconfig.json` is a file used by .NET 5+ applications to pass arbitrary configuration parameters via the application host to the .NET runtime.  On mobile and WebAssembly, we would like to support passing configuration properties from the Mono embedder to the runtime.
+
+To minimize the impact of the app startup time, the design should preserve these contraints:
+1. Don’t parse JSON file at runtime at application startup
+2. Don’t need to support full set of well-known properties, just the ones that make sense on mobile
+3. Don’t need to support shared framework configuration parameters
+4. Don't need to support the desktop host environment variables, startup hooks, or other configuration mechanisms that do not make sense on mobile.
+5. This is separate from the key/value set passed to `monovm_initialize`/`coreclr_initialize`  (ie we will not pass the additional properties via `monovm_initialize` - just using this new mechanism). Things like the TPA list, additional probing paths, PINVOKE_OVERRIDE, etc will be addressed by an evolution of `monovm_initialize` - https://github.com/dotnet/runtime/issues/48416
+
+## Design Overview
+
+This feature could be achived by mainly two parts:
+1. A new MSBuild task called `RuntimeConfigParser` will run after the `runtimeconfig.json` is created by the dotnet build process. The task will extract properties keys and values into a binary blob format. The resulting `runtimeconfig.blob` file will be bundled with the application.
+2. The runtime will expose a new API entrypoint `monovm_runtimeconfig_initialize` that gets either a path to pass to `mono_file_map_open` or a pointer to the blob in memory. Then, the runtime will read the binary data and populate the managed AppContext with the properties.
+
+We will only take the `runtimeOptions→configProperties` json key. Its content is a flat key-value string/boolean dictionary.
+
+The runtime will assume that the properties passed via `monovm_initialize` and `monovm_runtimeconfig_initialize` will be different. To ensure this, the MSBuild task that we will provide will be given a list of property names that the embedder promises it will pass to `monovm_initialize`. The MSBuild task will check that `runtimeconfig.json` does not set any of those same properties. If there is a duplicate, error out.
+
+We take everything and pass all the properties to the managed AppContext. For the well-known standard properties the mono runtime will read and propagate the ones it cares about and ignore the rest.
```

Comment:
```suggestion
We take everything and pass all the properties to the managed AppContext. For the well-known standard properties the mono runtime will read and use the ones it cares about.
```

(I think "and ignore the rest" makes it sound like we will drop them from the managed AppContex,t too.  Which is not the case)

Reply:
Fixed. Thanks for the suggestion!